FreeRTOS/FreeRTOS
# Expand RV32E demo support, prevent 'ebreak' looping

Description
-----------
The RegTest tests are modified in the Qemu and HiFive demos to support RV32E and its limited set of registers. This is done by using general purpose register x15 instead of x31 to compare against other registers in `vRegTest1Implementation`. Register x15 needs to be use as RV32E (or embedded) only has the first 16 general purpose registers that standard RV32/RV64 has.

Removes 'ebreak' call looping in RegTest. The `ebreak` instruction according to the RISC-V ASM manual is...

> 5.3.2 EBREAK
Environment Break (EBREAK) is an assembly instruction that is used to stop the execution suddenly.
Syntax
`ebreak`
Description
The EBREAK instruction is used to invoke a debugger, by causing a “Breakpoint” exception. Typically the debugging software will insert this instruction at various places in the application code sequence, in order to gain control from an executing program.

We don't need this to be called in a loop

Test Steps
-----------
1. Pull down this change.
2. Update submodule to point to [commit](https://github.com/FreeRTOS/FreeRTOS-Kernel/commit/1d59f650072e176a9ff1d7df1d5f9d9f17b75ab7) of the FreeRTOS-Kernel
3. Build HiFive demo (either IAR or GCC) for an RV32e processor. QEMU demo also works.
  a. Note: Building should be enough. IAR Embedded workbench (and likely other compilers) will fail to recognize the x16-x31 registers in assembly.
 4. Run the main-full demo.

Related Issue
-----------
Corrects demo issues due to changes made in https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/588


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
